tests/components/efergy/test_config_flow.py

```diff
@@ -0,0 +1,128 @@
+            user_input=CONF_DATA,
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == DEFAULT_NAME
+        assert result["data"] == CONF_DATA
+        assert result["result"].unique_id == HID
+
+
+async def test_flow_user_cannot_connect(hass: HomeAssistant):
+    """Test user initialized flow with unreachable service."""
+    with _patch_efergy_status() as efergymock:
+        efergymock.side_effect = exceptions.ConnectError
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={CONF_SOURCE: SOURCE_USER}, data=CONF_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "cannot_connect"
+
+
+async def test_flow_user_invalid_auth(hass: HomeAssistant):
+    """Test user initialized flow with invalid authentication."""
+    with _patch_efergy_status() as efergymock:
+        efergymock.side_effect = exceptions.InvalidAuth
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={CONF_SOURCE: SOURCE_USER}, data=CONF_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "invalid_auth"
+
+
+async def test_flow_user_unknown(hass: HomeAssistant):
+    """Test user initialized flow with unknown error."""
+    with _patch_efergy_status() as efergymock:
+        efergymock.side_effect = Exception
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={CONF_SOURCE: SOURCE_USER}, data=CONF_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "unknown"
+
+
+async def test_flow_import(hass: HomeAssistant):
+    """Test import step."""
+    with _patch_efergy(), _patch_setup():
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={CONF_SOURCE: SOURCE_IMPORT}, data=IMPORT_DATA
+        )
+
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == DEFAULT_NAME
+        assert result["data"] == CONF_DATA
+        assert result["result"].unique_id == HID
+
+
+async def test_flow_import_already_configured(hass: HomeAssistant):
+    """Test import step already configured."""
+    create_entry(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_IMPORT}, data=IMPORT_DATA
+    )
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_flow_reauth(hass: HomeAssistant):
+    """Test reauth step."""
+    entry = create_entry(hass)
+    with _patch_efergy(), _patch_setup():
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={CONF_SOURCE: SOURCE_REAUTH}, data=CONF_DATA
```

Comment:
There will be a unique id and entry id of the config entry set in the context when raising the ConfigEntryAuthFailed exception. Just for completeness it's good to include those too.